design/service-binding.md

```diff
@@ -0,0 +1,786 @@
+
+Here's the content of the ConfigMap, with type and protocol added:
+
+``` yaml
+type=kafka
+protocol=kafka
+endpoints.plain=myhost1.example.com:9092,myhost2.example.com:9092
+endpoints.tls=myhost3.example.com
+```
+
+And the `status` changed to point to it:
+
+``` yaml
+apiVersion:kafka.strimzi.io/v1beta1
+kind: Kafka
+metadata:
+  name: my-cluster
+spec:
+ ...
+status:
+  listeners:
+  - type: plain
+    addresses:
+    - host: myhost1.example.com
+      port: 9092
+    - host: myhost2.example.com
+      port: 9092
+  - type: tls
+    addresses:
+    - host: myhost3.example.com
+      port: 9093
+  bindingConfigMap: binding-cm
+```
+
+And the CSV annotations:
+
+``` yaml
+apiVersion: operators.coreos/com:v1alpha1
+kind: ClusterServiceVersion
+metadata:
+  name: strimzi-cluster-operator.v0.16.2
+  namespace: placeholder
+  annotations:
+    containerImage: docker.io/strimzi/operator:0.16.2
+spec:
+  customresourcedefinitions:
+    owned:
+    - description: Represents a Kafka cluster
+      displayName: Kafka
+      kind: Kafka
+      name: kafkas.kafka.strimzi.io
+      version: v1beta1
+      statusDescriptors:
+      - description: The binding information for Kafka clients
+        displayName: ConfigMap containing binding information
+        path: status.bindingConfigMap
+        x-descriptors:
+        - 'urn:alm:descriptor:servicebinding:configMap'
+```
+
+### Consuming client's ServiceBindingRequest
+
+All of the required annotations are applied to the `Kafka` CSV, so the binding should only need to refer to the `Kafka` CR.
+
+``` yaml
+apiVersion: service.binding/v1alpha1
+kind: ServiceBindingRequest
+metadata:
+  name: my-binding
+spec:
+  services:
+  - group: kafka.strimzi.io
+    kind: Kafka
+    version: v1beta1
+    resourceRef: my-cluster
+```
+
+The consuming client needs to know the listener name.
+
+## Binding to a Kafka cluster with TLS but no authentication
+
+The addition with this scenario is that Kafka clients need access to the CA certificate that signed the broker's server certificate. The Service Binding Specification does not currently have support for a separate CA certificate, which seems like a simple enhancement, which it indicated using `caSecret` in the example below.
+
+The CA certificate is most easily obtained from the `<cluster>-cluster-ca-cert` secret using the `ca.p12` and `ca.password` fields in the secret. To enable the Service Binding Operator to obtain this information, the same pattern of enhancing the CR `status` and annotating the CSV can be used.
+
+``` yaml
+apiVersion:kafka.strimzi.io/v1beta1
+kind: Kafka
+metadata:
+  name: my-cluster
+spec:
+ ...
+status:
+  listeners:
+  - type: plain
+    addresses:
+    - host: myhost1.example.com
+      port: 9092
+    - host: myhost2.example.com
+      port: 9092
+  - type: tls
+    addresses:
+    - host: myhost3.example.com
+      port: 9093
+  bindingConfigMap: binding-cm
+  caCertificateSecret:
+    secretName: my-cluster-cluster-ca-cert
```

Comment:
To keep it consistent with the `KafkaUser` CRD (ie `secret: my-user`), how about removing `secretName` and just using `caCertificateSecret: my-cluster-cluster-ca-cert`?

Reply:
That's a good point actually. Strimzi is a bit inconsistent in naming of secrets, but here it's more consistent to use what you suggested.